arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -1351,6 +1367,109 @@ void RocksDBEngine::prepareDropCollection(TRI_vocbase_t& /*vocbase*/,
   replicationManager()->drop(&coll);
 }
 
+arangodb::Result RocksDBEngine::dropCollectionFast(TRI_vocbase_t& vocbase,
+                                                   LogicalCollection& collection) {
+
+  auto* rocksdb = static_cast<RocksDBMetaCollection*>(collection.getPhysical());
+  bool const compactAfterDrop = rocksdb->meta().numberDocuments() >= 32 * 1024;
+
+  rocksdb::DB* db = _db->GetRootDB();
+
+  TRI_ASSERT(collection.status() == TRI_VOC_COL_STATUS_DELETED);
+
+  // Prepare collection remove batch
+  rocksdb::WriteBatch batch;
+  {
+    RocksDBLogValue logValue =
+        RocksDBLogValue::CollectionDrop(vocbase.id(), collection.id(),
+                                        arangodb::velocypack::StringRef(
+                                            collection.guid()));
+    batch.PutLogData(logValue.slice());
+  }
+  {
+    RocksDBKey key;
+    key.constructCollection(vocbase.id(), collection.id());
+    batch.Delete(RocksDBColumnFamilyManager::get(RocksDBColumnFamilyManager::Family::Definitions),
+                 key.string());
+  }
+
+  {
+    Result res =
+        RocksDBMetadata::deleteCollectionMetaBatch(db, rocksdb->objectId(), batch);
+    if (res.fail()) {
+      LOG_TOPIC("2c890", ERR, Logger::ENGINES)
+          << "error removing collection meta-data: " << res.errorMessage();  // continue regardless
+    }
+  }
+
+  // delete indexes, RocksDBIndex::drop() has its own check
+  std::vector<std::shared_ptr<Index>> vecShardIndex = rocksdb->getIndexes();
+  TRI_ASSERT(!vecShardIndex.empty());
+
+  for (auto& index : vecShardIndex) {
+    auto* ridx = static_cast<RocksDBIndex*>(index.get());
+    Result res = RocksDBMetadata::deleteIndexEstimateBatch(db, ridx->objectId(), batch);
+    if (res.fail()) {
+      LOG_TOPIC("f2d51", WARN, Logger::ENGINES)
+      << "could not delete index estimate: " << res.errorMessage();
+    }
+
+    auto dropRes = ridx->dropWithBatch(batch).errorNumber();
```

Comment:
`dropRes` is a full `Result` already, so we don't need to downgrade it into just an error number.
We could also recycle the existing `res` variable here, which is already in the scope.
```suggestion
    auto dropRes = ridx->dropWithBatch(batch);
```